garak/detectors/base.py

```diff
@@ -72,7 +72,10 @@ def detect(self, attempt: garak.attempt.Attempt) -> Iterable[float]:
 class HFDetector(Detector, HFCompatible):
     """Detector using a Hugging Face model"""
 
-    DEFAULT_PARAMS = Detector.DEFAULT_PARAMS | {"hf_args": {"device": "cpu"}}
+    DEFAULT_PARAMS = Detector.DEFAULT_PARAMS | {
+        "hf_args": {"device": "cpu"},
+        "tokenizer_kwargs": {"padding": True, "truncation": True, "max_length": 512},
```

Comment:
```suggestion
        "tokenizer_kwargs": {"padding": True, "truncation": True},
```

Reply:
The default here was selected to match the existing usage in `ToxicCommentModel` and `MustRefuteClaimModel`.  As noted `MustContradictNLI` does not consume these values and even sets some in local methods. `max_length` in `MustContradictNLI` is pushed down to `256`.

I could see moving the default into the extending classes with the models they match to. I would prefer exposing the values for config even if the default is not best for all possible injected config.

I will expand `MustContradictNLI` edits to model setting the values in the extended class and consuming them it the `detect` implementation there.

Reply:
```
% python -m garak --plugin_info detectors.misleading.MustContradictNLI
garak LLM vulnerability scanner v0.9.0.15.post1 ( https://github.com/leondz/garak ) at 2024-08-28T17:16:51.090783
Configured info on detectors.misleading.MustContradictNLI:
                        description: This detector is looking for the model to contradict a claim.
                     DEFAULT_PARAMS: {'skip': False, 'hf_args': {'device': 'cpu'}, 'tokenizer_kwargs': {'truncation': True, 'max_length': 256}, 'detector_model_path': 'ynie/roberta-large-snli_mnli_fever_anli_R1_R2_R3-nli', 'detector_target_class': 'label'}
                           accuracy: None
                             active: False
                              bcp47: en
                            doc_uri:
                           modality: {'out': ['text']}
                          precision: 0.0
                             recall: 0.0
                               tags: []
                           mod_time: 2024-08-28 22:12:23 +0000
```
```
% python -m garak --plugin_info detectors.misleading.MustRefuteClaimModel
garak LLM vulnerability scanner v0.9.0.15.post1 ( https://github.com/leondz/garak ) at 2024-08-28T17:17:06.537017
Configured info on detectors.misleading.MustRefuteClaimModel:
                        description: Using leondz/refutation_detector_distilbert to detect model refutations
                     DEFAULT_PARAMS: {'skip': False, 'hf_args': {'device': 'cpu'}, 'tokenizer_kwargs': {'padding': True, 'truncation': True, 'max_length': 512}, 'detector_model_path': 'leondz/refutation_detector_distilbert', 'detector_target_class': 'label'}
                           accuracy: 0.9506
                             active: True
                              bcp47: en
                            doc_uri:
                           modality: {'out': ['text']}
                          precision: 0.0
                             recall: 0.0
                               tags: []
                           mod_time: 2024-08-28 22:12:23 +0000
```

Reply:
aside: blank `doc_uri` is suspicious - `detectors.base.Detector.doc_uri = ""` and should be `None` - will update in separate PR